src/modules/roles/roles.service.ts

```
@@ -0,0 +1,196 @@
+        this.logger.log(
+          `Creating role: ${dto.name} with permissions: ${dto.permissionIds.join(', ')}`,
+        );
+        const { permissionIds, ...roleData } = dto;
+
+        // Fetch permissions within transaction
+        const permissions = await manager.getRepository(Permission).find({
+          where: { id: In(permissionIds) },
+        });
+
+        if (permissions.length !== permissionIds.length) {
+          this.logger.warn('Some permissions not found for IDs: ' + permissionIds.join(', '));
```

Biome (1.9.4)

[error] Template literals are preferred over string concatenation.

Unsafe fix: Use a template literal.


(lint/style/useTemplate)